File: Codes/ReverseInteger.cpp
class Solution {
public:
	int reverse(int x) {
		int i;
		long y = 0;
		if (x < 0) {
			x = -x;
			while (x != 0) {
				i = x % 10;
				y = 10 * y + i;
				x = (x - i) / 10;
			}
			y = -y;
		}
		else {
			while (x != 0) {
				i = x % 10;
				y = 10 * y + i;
				x = (x - i) / 10;
			}
		}
		if (y > INT_MAX) {
			y = INT_MAX;
		}
		return (int)y;
	}
};

File: Codes/ReverseInteger.py
class Solution:
    # @return an integer
    def reverse(self, x):
        y, z = 0, x
        if z < 0:
        	x = -x
        while x != 0:
        	y = y * 10 + x % 10
        	x = x // 10
        if z < 0:
        	y = -y
        return y